### PR TITLE
Type improvement

### PR DESCRIPTION
**Pull request by bluebill1049**

### CODE REVIEW

File: src/types/resolvers.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ export type Resolver<
   TFieldValues extends FieldValues = FieldValues,
   TContext extends object = object
 > = (
-  values: TFieldValues,
+  values: UnpackNestedValue<TFieldValues>,
```
Comment (kotarella1110) [inline src/types/resolvers.ts:24]: awesome 👍

**Review by kotarella1110 — APPROVED** [review]
LGTM 💯

File: src/useForm.ts [hunk 2]
```diff
@@ -1107,8 +1104,8 @@ export function useForm<
     dirtyFields,
   }: OmitResetState) => {
     if (!isValid) {
-      validFieldsRef.current = new Set();
-      fieldsWithValidationRef.current = new Set();
+      validFieldsRef.current = {};
```
[the remark this answers is not in the document]
Comment (bluebill1049) [reply src/useForm.ts:1107]: Thanks @pmaier983 the fix is actually correct, i have updated the type, it's not a `Set` but `object`

File: src/types/resolvers.ts (hunk 1 above)
Comment (bluebill1049) [reply src/types/resolvers.ts:24]: Thank you for the review :hear

Comment (bluebill1049) [reply src/types/resolvers.ts:24]: I will continuously improve our TS internally over time. 💪

**Comment by bluebill1049** [comment]
Thank you guys' review ❤️

**Pull request merged**